File: fingertip/src/main/java/com/praszapps/fingertip/view/FingertipApplication.java
package com.praszapps.fingertip.view;


import com.praszapps.fingertip.model.di.component.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

public abstract class FingertipApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends FingertipApplication> applicationInjector() {
        return DaggerAppComponent.builder().create(this);
    }
}

File: fingertip/src/main/java/com/praszapps/fingertip/view/FingertipApplication.kt
package com.praszapps.fingertip.view


import com.praszapps.fingertip.model.di.component.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication

abstract class FingertipApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out FingertipApplication> {
        return DaggerAppComponent.builder().create(this)
    }
}